tests/core/stack/test_stack.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+
+from evm.vm.stack import (
+    Stack,
+)
+from evm.exceptions import (
+    FullStack,
+    InsufficientStack,
+    ValidationError,
+)
+from evm.constants import (
+    UINT256,
+    BYTES,
+    SECPK1_N,
+)
+
+
+@pytest.fixture
+def stack():
+    return Stack()
+
+
+@pytest.mark.parametrize(
+    ("value,is_valid"),
+    (
+        (-1, False),
+        (0, True),
+        (1, True),
+        (2**256 - 1, True),
+        (2**256, False),
+        ('abcde', False),
+        (b'abcde', True),
+        (b'100100100100100100100100100100100', False),
+    )
+)
+def test_push_only_pushes_valid_stack_items(stack, value, is_valid):
+    if is_valid:
+        stack.push(value)
+        assert stack.values == [value]
+    else:
+        with pytest.raises(ValidationError):
+            stack.push(value)
+
+
+def test_push_does_not_allow_stack_to_exceed_1024_items(stack):
+    for num in range(1024):
+        stack.push(num)
+    assert len(stack.values) == 1024
+    with pytest.raises(FullStack):
+        stack.push(1025)
+
+
+def test_dup_does_not_allow_stack_to_exceed_1024_items(stack):
+    stack.push(1)
+    for num in range(1023):
+        stack.dup(1)
+    assert len(stack.values) == 1024
+    with pytest.raises(FullStack):
+        stack.dup(1)
+
+
+@pytest.mark.parametrize(
+    ("items,type_hint"),
+    (
+        ([1], UINT256),
+        ([1, 2, 3], UINT256),
+        ([b'1', b'10', b'101', b'1010'], BYTES)
```

Comment:
It would be good to have at least one of these for which the `type_hint` was different than the value being pushed onto the stack.

If it isn't clear, the `type_hint` tells the `stack` object what type it should return, so a good test would be pushing a `bytes` object onto the stack and pulling it back of as an `int`.